network/p2p/p2pnode/libp2pNode.go

```diff
@@ -532,7 +532,7 @@ func (n *Node) IsConnected(peerID peer.ID) (bool, error) {
 // SetRouting may be called at most once.
 func (n *Node) SetRouting(r routing.Routing) {
 	if n.routing != nil {
-		n.logger.Fatal().Msg("routing already set")
+		irrecoverable.Throw(context.TODO(), fmt.Errorf("routing already set"))
```

Comment:
I think it's better to stick with the fatal error unless there's a irrecoverable context to use.

Reply:
https://github.com/onflow/flow-go/pull/4800/commits/bb1aec2cf82d7a63eba50da45118e3fc46972a46